### PR TITLE
Added custom GlobalExceptionHandler for data validation:

### PR DESCRIPTION
**Pull request by GoldHolub**
- added method handleMethodArgumentNotValid in CustomGlobalExceptionHandler to catch all invalid input exceptions (@NotNull, @Min(0))
- added handleEntityNotFoundException method to catch all EntityNotFoundException
- added handleSqlIntegrityConstraintViolationException method to catch all cases connected with "Integrity Constraint Violation" like the same ISBN that should be unique for each book and so on.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,6 +72,12 @@
 			<version>${mapstruct.version}</version>
 		</dependency>
 
+		<dependency>
+			<groupId>org.hibernate</groupId>
+			<artifactId>hibernate-validator</artifactId>
```
Comment (Rommelua) [inline pom.xml]: use starter for dependesy

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
```
Comment (Rommelua) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java]: use NotBlank

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,13 +1,19 @@
 package com.example.demo.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotBlank
```
Comment (sokimaaa) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java:12]: Does NotBlank prevent null-value?

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
     private String title;
+    @NotBlank
     private String author;
+    @NotBlank
```
Comment (sokimaaa) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java]: consider extra annotations, e.g. @ISBN

**Pull request merged**